src/stream/src/task/barrier_manager/managed_state.rs

```diff
@@ -185,39 +151,37 @@ impl ManagedBarrierState {
         actor_ids_to_collect: impl IntoIterator<Item = ActorId>,
         collect_notifier: oneshot::Sender<CollectResult>,
     ) {
-        match self.inner_mut() {
-            ManagedBarrierStateInner::Pending { .. } => {
-                let remaining_actors = actor_ids_to_collect.into_iter().collect();
-
-                *self.inner_mut() = ManagedBarrierStateInner::Issued {
-                    epoch: barrier.epoch.curr,
-                    remaining_actors,
-                    collect_notifier,
-                };
-                self.may_notify();
-            }
-
-            ManagedBarrierStateInner::Stashed {
-                epoch,
-                collected_actors,
-            } => {
-                assert_eq!(barrier.epoch.curr, *epoch);
-
+        match self.map.get(&barrier.epoch.curr) {
```

Comment:
May use entry API here. :)
```suggestion
        match self.map.entry(barrier.epoch.curr) {
```

Reply:
okok

Reply:
I try to use entry, But I find it's not getting simpler. So I use get with some change